eval! doesn't compile on the latest nightly

# Versions
Rust:  rustc 1.52.0-nightly (07194ffcd 2021-02-10)
`glsp`: both 0.2.0 and the latest master (3fd194a)

# Minimal reproduction example
```
use glsp::prelude::*;
fn main() {
    let _: Val = eval!("(+ 1 2)").unwrap();
}
```

No matter the input, I can't get `eval!` to compile, instead I always encounter the following error:

```
error[E0716]: temporary value dropped while borrowed
   --> src/main.rs:193:26
    |
193 |         let _: GlspVal = eval!("1.0").unwrap();
    |                          ^^^^^^^^^^^-
    |                          |          |
    |                          |          temporary value is freed at the end of this statement
    |                          creates a temporary which is freed while still in use
    |                          argument requires that borrow lasts for `'static`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0597]: `__glsp_eval_args` does not live long enough
   --> src/main.rs:193:26
    |
193 |         let _: GlspVal = eval!("1.0").unwrap();
    |                          ^^^^^^^^^^^-
    |                          |          |
    |                          |          `__glsp_eval_args` dropped here while still borrowed
    |                          borrowed value does not live long enough
    |                          argument requires that `__glsp_eval_args` is borrowed for `'static`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```
Running through `cargo expand` (and massaging the expanded code a little):
```
fn main() {
    let _: Val = ::glsp::with_lazy_val(
        "src/main.rs:4:18",
        || ::glsp::load_compiled(b"many bytecode").unwrap(),
        |__glsp_eval_val| {
            let __glsp_eval_args = ::glsp::GResult::Ok(::glsp::arr())?;
            let __glsp_eval_result: ::glsp::Val = match __glsp_eval_val {
                ::glsp::Val::GFn(gfn) => ::glsp::call(gfn, &[&__glsp_eval_args])?,
                _ => panic!(),
            };
            ::glsp::FromVal::from_val(&__glsp_eval_result)
        },
    )
    .unwrap();
}
```

And the error:

```
error[E0716]: temporary value dropped while borrowed
  --> src/main.rs:30:61
   |
30 |                 ::glsp::Val::GFn(gfn) => ::glsp::call(gfn, &[&__glsp_eval_args])?,
   |                                          -------------------^^^^^^^^^^^^^^^^^^^-- temporary value is freed at the end of this statement
   |                                          |                  |
   |                                          |                  creates a temporary which is freed while still in use
   |                                          argument requires that borrow lasts for `'static`

error[E0597]: `__glsp_eval_args` does not live long enough
  --> src/main.rs:30:62
   |
30 |                 ::glsp::Val::GFn(gfn) => ::glsp::call(gfn, &[&__glsp_eval_args])?,
   |                                          --------------------^^^^^^^^^^^^^^^^^--
   |                                          |                   |
   |                                          |                   borrowed value does not live long enough
   |                                          argument requires that `__glsp_eval_args` is borrowed for `'static`
...
34 |         },
   |         - `__glsp_eval_args` dropped here while still borrowed

```

Experimenting a little, it looks like removing the inner borrow of `__glsp_eval_args` makes the borrow checker happy.

--- a/glsp-proc-macros2/src/lib.rs
+++ b/glsp-proc-macros2/src/lib.rs
@@ -304,7 +304,7 @@ pub fn eval(input: TokenStream) -> TokenStream {
                     |__glsp_eval_val| {
                         let __glsp_eval_args = ::glsp::try_arr![#(&#vars_to_read),*]?;
                         let __glsp_eval_result: ::glsp::Val = match __glsp_eval_val {
-                            ::glsp::Val::GFn(gfn) => ::glsp::call(gfn, &[&__glsp_eval_args])?,
+                            ::glsp::Val::GFn(gfn) => ::glsp::call(gfn, &[__glsp_eval_args])?,
                             _ => ::std::panic!()
                         };
                         #(

